### PR TITLE
Morphism from fiber product to sink morphism from source to pushout

### PR DESCRIPTION
**Pull request by mohamed-barakat**
Added support for
* MorphismFromFiberProductToSink
* MorphismFromSourceToPushout

### CODE REVIEW

**Comment by zickgraf** [comment]
You should rebase this on master to fix the CI failures.

**Comment by mohamed-barakat** [comment]
@zickgraf: Done, thanks.

**Comment by mohamed-barakat** [comment]
@zickgraf: Why is this PR failing?

**Comment by zickgraf** [comment]
As far as I can see, CircleCI is not failing (the second commit 6f47cdb3de55288f366d83bda51fc9b04693012f has a green tick). However, the code coverage is decreasing and thus the Codecov Report is red.

File: CAP/gap/MethodRecord.gi [hunk 1]
```diff
@@ -1493,6 +1493,66 @@ ProjectionInFactorOfFiberProductWithGivenFiberProduct := rec(
...
 
+MorphismFromFiberProductToSink := rec(
+  installation_name := "MorphismFromFiberProductToSinkOp",
+  argument_list := [ 1 ],
+  filter_list := [ IsList, "morphism" ],
+  io_type := [ [ "D" ], [ "P", "D_1_range" ] ],
+  cache_name := "MorphismFromFiberProductToSinkOp",
+  universal_object_position := "Source",
+  universal_type := "Limit",
+  #dual_operation := "MorphismFromSourceToPushout",
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/MethodRecord.gi]: I probably forgot to uncomment it in the next commit 6f47cdb.

File: CAP/gap/MethodRecord.gi [hunk 2]
```diff
@@ -1907,6 +1907,66 @@ InjectionOfCofactorOfPushoutWithGivenPushout := rec(
   end,
   return_type := "morphism" ),
 
+MorphismFromSourceToPushout := rec(
+  installation_name := "MorphismFromSourceToPushoutOp",
+  argument_list := [ 1 ],
+  filter_list := [ IsList, "morphism" ],
+  io_type := [ [ "D" ], [ "D_1_range", "I" ] ],
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/MethodRecord.gi]: Stupid copy-paste mistake.

File: CAP/gap/MethodRecord.gi [hunk 3]
```diff
@@ -1907,6 +1907,66 @@ InjectionOfCofactorOfPushoutWithGivenPushout := rec(
...
+    od;
+    
+    return [ true ];
+  end,
+  return_type := "morphism" ),
+
+MorphismFromSourceToPushoutWithGivenPushout := rec(
+  installation_name := "MorphismFromSourceToPushoutWithGivenPushout",
+  filter_list := [ IsList, "object" ],
+  io_type := [ [ "D", "I" ], [ "D_1_range", "I" ] ],
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/MethodRecord.gi]: Stupid copy-paste mistake.

File: CAP/gap/DerivedMethods.gi [hunk 4]
```diff
@@ -262,6 +274,18 @@ AddWithGivenDerivationPairToCAP( InjectionOfCofactorOfPushout,
     
   end : Description := "InjectionOfCofactorOfPushout by composing the direct sum injection with the direct sum projection to the pushout" );
 
+##
+AddWithGivenDerivationPairToCAP( MorphismFromSourceToPushout,
+  
+  function( diagram )
+    local iota_1;
+    
+    iota_1 := InjectionOfCofactorOfDirectSum( diagram, 1 );
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply CAP/gap/DerivedMethods.gi]: Yes, thanks. Another a copy-paste-error.

**Comment by mohamed-barakat** [comment]
I'll add the requested tests later an then inform you here.

**Comment by sebastianpos** [comment]
ping (for the desired test file)

**Comment by mohamed-barakat** [comment]
I am happy with the example, thank you very much Sebastian :)

**Pull request merged**